extension/oauth2authextension/extension.go

```diff
@@ -0,0 +1,90 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oauth2authextension
+
+import (
+	"context"
+	"net/http"
+
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"google.golang.org/grpc/credentials"
+	grpcOAuth "google.golang.org/grpc/credentials/oauth"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configauth"
+)
+
+// OAuth2Authenticator provides implementation for providing client authentication using OAuth2 client credentials
```

Comment:
Go conventions always confuse me. Should this be `Oauth2Authenticator`?

@rakyll, I noticed that you recently sent a PR fixing a few components that weren't following the conventions. Perhaps you can enlighten us here?

Reply:
It should be OAuth2Authenticator.